ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/Replace.java

```diff
@@ -45,4 +48,46 @@ public String replace(
 
     return str.replace(oldStr, newStr);
   }
+
+  @Udf
+  public ByteBuffer replace(
+      @UdfParameter(
+          description = "The source bytes."
+              + " If null, then function returns null.") final ByteBuffer source,
+      @UdfParameter(
+          description = "The byte subset to replace."
+              + " If null, then function returns null.") final ByteBuffer toReplace,
+      @UdfParameter(
+          description = "The bytes to replace the old bytes with."
+              + " If null, then function returns null.") final ByteBuffer replaceWith) {
+
+    if (source == null || toReplace == null || replaceWith == null) {
+      return null;
+    }
+
+    //if replaceWith.length == toReplace.length, DON'T create a new array
+    //else, create an arraylist. Create another arraylist2 and a value matchingBytesNum.
+    //populate arraylist2 as long as the next byte matches the next byte in our toReplace[matchingBytesNum]
+    //if matchingBytesNum reaches the length of toReplace, we have a matching subbytes and append
+    //to our 1st arraylist the replaceWith bytes. If they diverge, just dump the original arraylist
```

Comment:
This was how I was going to solve the problem with a custom solution if we decide to go down that route (please view my comment in the discussion section to understand my options). I'll leave this here. AFAICT the solution is O(N), but it's probably slower and more difficult to read than others in the end.